src/components/PostSelector/PostSelector.js

```diff
@@ -0,0 +1,236 @@
+import { isEmpty } from 'lodash'
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import useInView from 'react-cool-inview'
+import { useDrag, useDrop } from 'react-dnd'
+import { useDispatch, useSelector } from 'react-redux'
+import update from 'immutability-helper'
+import { FETCH_POSTS } from 'store/constants'
+import Icon from 'components/Icon'
+import Loading from 'components/Loading'
+import RoundImage from 'components/RoundImage'
+import useDebounce from 'hooks/useDebounce'
+import { fetchPosts } from 'routes/Stream/Stream.store'
+import isPendingFor from 'store/selectors/isPendingFor'
+
+import './PostSelector.scss'
+
+const PAGE_SIZE = 10
+
+export default function PostSelector ({ collection, draggable, group, onRemovePost, onReorderPost, onSelectPost, posts }) {
+  const dispatch = useDispatch()
+  const [autocomplete, setAutocomplete] = useState('')
+  const [suggestions, setSuggestions] = useState([])
+  const [offset, setOffset] = useState('')
+  const [selectedPosts, setSelectedPosts] = useState(posts || [])
+  const [suggestionsOpen, setSuggestionsOpen] = useState(false)
+  const searchBox = useRef(false)
+
+  const debouncedAutcomplete = useDebounce(autocomplete, 300)
+
+  useEffect(() => setSelectedPosts(posts || []), [posts])
+
+  useEffect(() => {
+    // TODO: tell people they need to type 2 characters to get results?
+    if (!autocomplete?.length || autocomplete.length > 1) {
+      dispatch(fetchPosts({ collectionToFilterOut: collection?.id, context: 'groups', slug: group.slug, search: debouncedAutcomplete, sortBy: 'created', first: PAGE_SIZE, offset: 0 })).then(res => {
```

Comment:
I'd prefer to see this broken down into lines.

Reply:
Also could make this a an async `useEffect(...` and do `const response = await dispatch(fetchPosts({ ...`. I woudl prefer that as I find it more readable, but there may be something I'm missing about the need to do this as a traditional Promise here.

Reply:
i can probably do that